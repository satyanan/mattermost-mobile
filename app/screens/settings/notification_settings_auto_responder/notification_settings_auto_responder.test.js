// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import Preferences from 'mattermost-redux/constants/preferences';
import {General} from 'mattermost-redux/constants';
import {shallowWithIntl} from 'test/intl-test-helper';

import NotificationSettingsAutoResponder from './notification_settings_auto_responder';
import TextInputWithLocalizedPlaceholder from 'app/components/text_input_with_localized_placeholder';
import Section from 'app/screens/settings/section';

describe('NotificationSettingsAutoResponder', () => {
    const navigator = {
        setOnNavigatorEvent: jest.fn(),
        setButtons: jest.fn(),
    };

    const baseProps = {
        currentUser: {},
        theme: Preferences.THEMES.default,
        onBack: jest.fn(),
        currentUserStatus: General.OUT_OF_OFFICE,
        updateMeRequest: {},
        enableOutOfOfficeDatePicker: false,
        navigator,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <NotificationSettingsAutoResponder {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.setProps({enableOutOfOfficeDatePicker: false}));
        expect(wrapper.find(TextInputWithLocalizedPlaceholder).length).toBe(0);
        expect(wrapper.find(Section).length).toBe(1);
    });
});
