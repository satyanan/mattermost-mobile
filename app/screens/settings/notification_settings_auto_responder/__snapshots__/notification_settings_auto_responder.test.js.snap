// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettingsAutoResponder should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "backgroundColor": "rgba(61,60,64,0.06)",
        "flex": 1,
        "paddingTop": 35,
      }
    }
  >
    <section
      disableHeader={true}
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
    >
      <sectionItem
        action={[Function]}
        actionType="toggle"
        label={
          <FormattedText
            defaultMessage="Enabled"
            id="mobile.notification_settings.auto_responder.enabled"
          />
        }
        selected={false}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
    </section>
    <FormattedText
      defaultMessage="Set a custom message that will be automatically sent in response to Direct Messages. Mentions in Public and Private Channels will not trigger the automated reply. Enabling Automatic Replies sets your status to Out of Office and disables email and push notifications."
      id="mobile.notification_settings.auto_responder.footer_message"
      style={
        Object {
          "color": "rgba(61,60,64,0.5)",
          "fontSize": 12,
          "marginHorizontal": 15,
        }
      }
    />
  </View>
</View>
`;
